### PR TITLE
change monospace font stack

### LINKED ISSUE #179: Add "Operator Mono" as <pre>'s default font
image

### PR DESCRIPTION
**Pull request by jacobbearden**
closes #179
I believe I covered all style selectors that set the normal monospace stack.

### CODE REVIEW

File: extension/custom.css [hunk 1]
```diff
@@ -11,3 +11,38 @@ a[href^="github-mac"] {
...
+pre,
+tt,
+.ace_editor.ace-github-light,
+.blame-sha,
+.blob-code-inner,
+.blob-num,
+.branch-name,
+.commit .sha,
+.commit .sha-block,
+.commit-desc pre,
```
Comment (paulmolluzzo) [inline extension/custom.css]: Is this necessary with the other `pre` listed above? I would think the `!important` would be enough to enforce the style on all `pre` tags even if they're selected more specifically, unless GitHub is putting `!important` in their CSS (I'd be surprised to see that).

If it's required leave it, of course.

Same comment applies to the couple of other scenarios below.

**Comment by paulmolluzzo** [comment]
@jacobbearden thanks! I made a small comment in the code, but otherwise this looks good in all the places I checked (code files, comments, commit messages, sha blocks).

// @sindresorhus @DrewML @sorrycc

File: extension/custom.css (hunk 1 above)
Comment (jacobbearden) [inline extension/custom.css]: Probably not necessary, I just searched the css for anywhere the `Consolas, "Liberation Mono", Menlo, Courier, monospace` stack was used, and copy pasted the selector. I can remove the redundant ones if you would like.

Comment (paulmolluzzo) [inline extension/custom.css]: Let's leave it.

**Comment by paulmolluzzo** [comment]
LGTM :+1: Thanks!

File: extension/custom.css [hunk 2]
```diff
@@ -11,3 +11,38 @@ a[href^="github-mac"] {
...
+.input-monospace,
+.news .alert .branch-link,
+.oauth-app-info-container dl.keys dd,
+.quick-pull-choice .new-branch-name-input input,
+.tag-references>li.commit,
+.two-factor-secret,
+.upload-files .filename,
+.url-field,
+.wiki-wrapper .wiki-history .commit-meta code {
+	font-family: "Source Code Pro", "Droid Sans Mono", "Operation Mono", Consolas, "Liberation Mono", Menlo, Courier, monospace !important;
```
Comment (sindresorhus) [inline extension/custom.css]: Use single-quotes

Comment (sindresorhus) [inline extension/custom.css]: I would also change the order to list the least popular first

Comment (sindresorhus) [inline extension/custom.css]: Order: droid sans mono, operation mono, liberation mono, source code pro, menlo, consolas, courier, monospace

Comment (sindresorhus) [inline extension/custom.css]: Also add a comment about the ordering. That it's sorted by least popular so you're most likely to get the font you want.

**Comment by jacobbearden** [comment]
@sindresorhus updated!

**Comment by paulmolluzzo** [comment]
Nice! Thanks @jacobbearden!

**Comment by pzi** [comment]
Hello,

FWIW, as the chrome extension got updated for me yesterday, I noticed the font change and it was, IMO, quite frustrating...

Comparison:
image

I have Menlo and Source Code Pro fonts installed, but to me, this change meant either not using this Chrome extensions any more or installing the fonts in the new stack. That's how much it affected me.

I am all for refining github, however, given the source code is one of _the_ main features of github and probably the one thing I spend most of my time looking at (reviewing pull requests, mainly), this was quite a big change.

This made me wonder:

1) how much refining is too much?
2) could there be a setting as to which font someone prefers? Seems like a very opinionated thing (I for myself use Ubuntu Mono Regular)
3) Should there be a mention in the README about the font stack with pointers to what the font options are and where to get them?

Would love to hear what others think about the above. Maybe I am the only one having this issue 😅 

Cheers & thanks
Patrik

**Comment by paulmolluzzo** [comment]
@pzi since this PR is already merged, you'd probably get more visibility with a new issue, if you're so inclined.

> 1) how much refining is too much?

People have asked about adding options (See #193, it might be the most recent discussion). For now it's sort of take it or leave it because there are no "options" available to play with, but you can [modify the code](https://github.com/sindresorhus/refined-github#customization) and [install manually](http://superuser.com/a/247654/6877).

2) could there be a setting as to which font someone prefers? Seems like a very opinionated thing (I for myself use Ubuntu Mono Regular)

You can make a PR with `Ubuntu Mono Regular` as the first font. 😄  Then it'll come up for you and anyone that has that font installed.

3) Should there be a mention in the README about the font stack with pointers to what the font options are and where to get them?

That wouldn't hurt either I guess.

**Comment by sindresorhus** [comment]
Lol. I knew someone would complain about this.

> I have Menlo and Source Code Pro fonts installed

Why do you have those installed if you don't like them? What really is the problem here?

> 1) how much refining is too much?

We are power users of GitHub and refine to how we like GitHub to be. We're upfront about this being a highly opinionated plugin. That's the only way you can really improve things. If enough people complain about something we always reconsider.

> 2) could there be a setting as to which font someone prefers? Seems like a very opinionated thing (I for myself use Ubuntu Mono Regular)

We're not interested in options, but happy to add more fonts to the list, like `Ubuntu Mono Regular`.

> 3) Should there be a mention in the README about the font stack with pointers to what the font options are and where to get them?

Sure

**Comment by pzi** [comment]
>Why do you have those installed if you don't like them?

Menlo is a system font cannot disable that one. Source Code Pro... not sure, maybe from when I was experimenting with monospace fonts to use for source code :)

---

Either way, thanks for getting back to me this quickly.

**Pull request merged**